tests/src/Functional/AccessTest.php

```diff
@@ -0,0 +1,251 @@
+        }
+        $users = [
+            'anonymous' => new AnonymousUserSession(),
+            'authenticated' => $this->createUser(),
+            'reader' => $this->createUser([
+                'preview document in collabora',
+            ]),
+            'editor' => $this->createUser([
+                'preview document in collabora',
+                'edit any document in collabora',
+            ]),
+            // Usually, a user with write access would also be given read access.
+            'writer' => $this->createUser([
+                'edit any document in collabora',
+            ]),
+            'diary keeper' => $diary_keeper,
+        ];
+
+        // Both routes are only accessible for admin.
+        $this->assertSameReport(
+            [
+                '/cool/view/<document>' => ['reader', 'editor'],
+                '/cool/edit/<document>' => ['editor', 'writer'],
+                '/cool/view/<wiki>' => ['anonymous'],
+                '/cool/edit/<wiki>' => ['anonymous'],
+                '/cool/view/<announcement>' => ['anonymous'],
+                '/cool/edit/<announcement>' => [],
+                '/cool/view/<own diary>' => ['diary keeper'],
+                '/cool/edit/<own diary>' => ['diary keeper'],
+                '/cool/view/<other diary>' => ['diary keeper'],
+                '/cool/edit/<other diary>' => [],
+            ],
+            $this->buildAccessReport($paths, $users),
+            'Users with access to given paths',
+        );
+    }
+
+    /**
+     * Builds a report about which users can access a given content.
+     *
+     * @param list<string>|array<string, string> $paths
+     *   A list of paths, optionally keyed.
+     * @param array<string, \Drupal\Core\Session\AccountInterface> $users
+     *   Accounts to test access with.
+     *
+     * @return array
+     *   Report to compare against in assertions.
+     */
+    protected function buildAccessReport(array $paths, array $users): array {
+        $report = [];
+        foreach ($paths as $path_key => $path) {
+            if (is_numeric($path_key)) {
+                $path_key = $path;
+            }
+            $report[$path_key] = [];
+            foreach ($users as $user_key => $user) {
+                $url = Url::fromUserInput($path);
+                $access = $url->access($user);
+                if ($access) {
+                    $report[$path_key][] = $user_key;
+                }
+            }
+        }
+        return $report;
+    }
+
+    /**
+     * Creates a media entity with attached file.
+     *
+     * @param string $type
+     *   Media type.
+     * @param array $values
+     *   Values for the media entity.
+     *
+     * @return \Drupal\media\MediaInterface
+     *   New media entity.
+     */
+    protected function createMediaEntity(string $type, array $values = []): MediaInterface {
+        file_put_contents('public://test.txt', 'Hello test');
```

Comment:
use \Drupal::service('file_system')->copy(

Reply:
We are not copying, we just put file content.. to copy we have to put an example file. or maybe use one that already exists.

Reply:
yes, I would go with the whole sh'bang, put a fixtures folder under test etc etc

Reply:
I actually find a bunch of `file_put_contents('public://' .` in core tests.
I would prefer to keep this simple snippet unless you can point out what could possibly go wrong.